modules/meshrenderinggl/glsl/illustration/sortandfill.frag

```diff
@@ -0,0 +1,114 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+/**
+ * Convertion from the a-buffer (fragment lists) to the illustration buffers:
+ * Load fragments, sort them, store them in linear memory
+ */
+
+// this is important for the occlusion query
+layout(early_fragment_tests) in;
+
+#include "oit/abufferlinkedlist.glsl"
+#include "illustration/illustrationbuffer.glsl"
+#include "oit/sort.glsl"
+
+// atomic counter to allocate space in the illustration buffer
+// just reuse the counter from the A-Buffer
+#define illustrationBufferCounter abufferCounter
+
+// We need some way to add an arbitrary count to the atomic buffer, not just increment it
+#if defined(GLSL_VERSION_460)
+#define atomicAdd atomicCounterAdd
+#elif defined(GL_ARB_shader_atomic_counter_ops)
+#define atomicAdd atomicCounterAddARB
+#else
+#error \
+    "Sorry, but to fill the Illustration Buffer, I need to be able to add an atomic counter: OpenGL 4.6 or ARB_shader_atomic_counter_ops"
```

Comment:
"OpenGL 4.6 or ARB_shader_atomic_counter_ops is required to fill the Illustration Buffer using atomic counter."